File: src/test/java/com/sfh/pokeRogueBot/file/FileManagerTest.java
package com.sfh.pokeRogueBot.file;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;
import static org.mockito.internal.util.Platform.OS_NAME;

class FileManagerTest {

    boolean isWindows() {
        return OS_NAME.toLowerCase().contains("win");
    }

    boolean isMac() {
        return OS_NAME.toLowerCase().contains("mac");
    }

    FileManager fileManager = new FileManager();

    @Test
    void a_filepath_is_returned(){

        String fileNamePrefix = "test";
        String filePath = fileManager.getTempFilePath(fileNamePrefix);

        if(isWindows()){
            assertEquals(".\\data\\temp\\0_test.png", filePath);
        } else if(isMac()){
            assertEquals("./data/temp/0_test.png", filePath);
        } else {
            fail("Unknown OS, please add the OS to the test");
        }
    }

    @Test
    void a_directory_path_is_returned(){

        String screenshotTempDirPath = fileManager.getScreenshotTempDirPath();

        if(isWindows()){
            assertEquals(".\\data\\temp\\", screenshotTempDirPath);
        } else if(isMac()){
            assertEquals("./data/temp/", screenshotTempDirPath);
        } else {
            fail("Unknown OS, please add the OS to the test");
        }
    }

}

File: src/test/kotlin/com/sfh/pokeRogueBot/file/FileManagerTest.kt
package com.sfh.pokeRogueBot.file

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.fail

class FileManagerTest {

    private fun isWindows(): Boolean {
        return System.getProperty("os.name").lowercase().contains("win")
    }

    private fun isMacOrLinux(): Boolean {
        val osName = System.getProperty("os.name").lowercase()
        return osName.contains("mac") || osName.contains("linux") || osName.contains("nix")
    }

    private val fileManager = FileManager()

    @Test
    fun `a filepath is returned`() {
        val fileNamePrefix = "test"
        val filePath = fileManager.getTempFilePath(fileNamePrefix)

        when {
            isWindows() -> {
                assertEquals(".\\data\\temp\\0_test.png", filePath)
            }

            isMacOrLinux() -> {
                assertEquals("./data/temp/0_test.png", filePath)
            }

            else -> {
                fail("Unknown OS: ${System.getProperty("os.name")}, please add the OS to the test")
            }
        }
    }

    @Test
    fun `a directory path is returned`() {
        val screenshotTempDirPath = fileManager.screenshotTempDirPath

        when {
            isWindows() -> {
                assertEquals(".\\data\\temp\\", screenshotTempDirPath)
            }

            isMacOrLinux() -> {
                assertEquals("./data/temp/", screenshotTempDirPath)
            }

            else -> {
                fail("Unknown OS: ${System.getProperty("os.name")}, please add the OS to the test")
            }
        }
    }
}